messaging/tests/test_messaging.py

```diff
@@ -188,20 +188,21 @@ def test_recv_one(self):
     self.assertIsInstance(recvd, capnp._DynamicStructReader)
     assert_carstate(msg.carState, recvd.carState)
 
+  unittest.skipIf("ZMQ" in os.environ, "send() is not blocking on ZMQ, causing test to fail")
```

Comment:
the test should still run, but expect failure if running with ZMQ. that way we know if the behavior changes

Reply:
Since built-in `expectedFailure` decorator does not support conditions, I've added custom `zmq_expected_failure` which applies it only in ZMQ mode